### PR TITLE
Remove obsolete `GridSearch` code

### PR DESCRIPTION
**Pull request by riedgar-ms**
With the `Moment`-based implementation of `GridSearch` in, remove obsolete code:
- Handwritten reweighting/relabelling in `GridSearch`
- Portions of `metrics` such as `metrics.DemographicParity` which are just used as `enum`s

### CODE REVIEW

File: notebooks/Grid Search for Binary Classification.ipynb [hunk 1]
```diff
@@ -208,8 +208,9 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "lagrange_multipliers = [x.lagrange_multiplier for x in (first_sweep.all_results)]\n",
-    "lagrange_multipliers"
+    "lagrange_multipliers = [x.lagrange_multiplier.iat[0] - x.lagrange_multiplier.iat[2] for x in (first_sweep.all_results)]\n",
+    "lagrange_multipliers\n",
+    "first_sweep.best_result.lagrange_multiplier#[(\"+\", \"all\", 2)]"
```
Comment (riedgar-ms) [inline notebooks/Grid Search for Binary Classification.ipynb]: This is going to need some explanation, and I also need to finish updating this notebook for the new Lagrange multiplier style

File: notebooks/Grid Search with Census Data.ipynb [hunk 2]
```diff
@@ -203,10 +203,10 @@
    "outputs": [],
    "source": [
     "sweep.fit(X_train, Y_train,\n",
-    "          aux_data=A_train,\n",
-    "          number_of_lagrange_multipliers=71)\n",
+    "          aux_data=A_train)\n",
     "\n",
-    "models = [ z.model for z in sweep.all_results]"
+    "models = [ z.model for z in sweep.all_results]\n",
+    "len(models)"
```
Comment (riedgar-ms) [inline notebooks/Grid Search with Census Data.ipynb]: Oops.... should delete

File: fairlearn/reductions/grid_search/grid_search.py [hunk 3]
```diff
@@ -125,14 +117,6 @@ def fit(self, X, y, aux_data=None, **kwargs):
...
-        lagrange_multipliers = None
-        if self._KW_LAGRANGE_MULTIPLIERS in kwargs:
-            lagrange_multipliers = kwargs[self._KW_LAGRANGE_MULTIPLIERS]
-
-        number_of_lagrange_multipliers = None
-        if self._KW_NUMBER_LAGRANGE_MULTIPLIERS in kwargs:
-            number_of_lagrange_multipliers = kwargs[self._KW_NUMBER_LAGRANGE_MULTIPLIERS]
-
         # Extract the target attribute
         A = self._make_vector(aux_data, "aux_data")
```
Comment (romanlutz) [inline fairlearn/reductions/grid_search/grid_search.py]: aux_data?

Comment (romanlutz) [reply fairlearn/reductions/grid_search/grid_search.py]: also, we may not want to call it A anymore

Comment (riedgar-ms) [reply fairlearn/reductions/grid_search/grid_search.py]: Internally, we can call it A, and there's more refactoring to come. As for renaming `aux_data` …. let's get agreement first

**Pull request merged**